src/reduction.jl

```diff
@@ -2,14 +2,15 @@
 
 # The entry format is (cudaname, julianame, merge, item, init)
 # ai is the accumulator, xi is the array element
-if VERSION >= v"0.6.0"
-    import AutoGrad: sumabs_, sumabs2_, minabs_, maxabs_
-    Base.sum(::typeof(abs), x::KnetArray, d...) = sumabs_(x,d...);
-    Base.sum(::typeof(abs2), x::KnetArray, d...) = sumabs2_(x,d...);
-    Base.maximum(::typeof(abs), x::KnetArray, d...) = maxabs_(x,d...);
-    Base.minimum(::typeof(abs), x::KnetArray, d...) = minabs_(x,d...);
-    reduced_dims_compat(dims,region)=map(last, Base.reduced_indices(map(Base.OneTo, dims), region))
-    reduction_ops = [
+import AutoGrad: sumabs_, sumabs2_, minabs_, maxabs_
```

Comment:
Any reason things changed here?  Did you just clear the pre-0.6 pieces?  Let's try to keep the changes to a relevant minimum so it is easy to review/test...

Reply:
Just clear pre-0.6, I cannot resist removing all those version check when I see them, sorry